airbyte-webapp/src/packages/cloud/views/auth/SignupPage/components/SignupForm.tsx

```diff
@@ -195,15 +196,27 @@ export const SignupForm: React.FC = () => {
     return yup.object().shape(shape);
   }, [showName, showCompanyName]);
 
-  return (
-    <Formik<FormValues>
-      initialValues={{
+  const [params] = useSearchParams();
+  const search = Object.fromEntries(params);
+
+  const initialValues = params.get("email")
+    ? {
+        name: search.firstname ? `${search.firstname} ${search.lastname}` : "",
+        companyName: search.company,
+        email: search.email,
+        password: "",
+        news: true,
+      }
+    : {
```

Comment:
Can be shorter + we protected from setting `undefined` as a value for fields
```suggestion
  const initialValues = {
    name: `${search.firstname ?? ""} ${search.lastname ?? ""}`.trim(),
    companyName: search.company ?? "",
    email: search.email ?? "",
    password: "",
    news: true,
  };
```

Reply:
Sounds good!